### PR TITLE
fix: remove enforcing the targeting key to be present

### PR DESCRIPTION
**Pull request by vahidlazio**
Remove the check for having the targeting key as the field is optional in the backend, prioritise the targeting key from the context protocol

### CODE REVIEW

File: src/test/java/com/spotify/confidence/FeatureProviderTest.java [hunk 1]
```diff
@@ -96,19 +97,16 @@ static void after() {
 
   @Test
   public void missingTargetKey() {
```
Comment (andreas-karlsson) [inline src/test/java/com/spotify/confidence/FeatureProviderTest.java:99]: Maybe this test should just be removed? Or reimplemented as a resolve test with missing targeting key. Right now the resolve fails with deadline exceeded, so the test takes some time and it's confusing why it evaluates to an error-case.

File: src/test/java/com/spotify/confidence/FeatureProviderTest.java [hunk 2]
```diff
@@ -96,19 +97,16 @@ static void after() {
...
 
-    assertThat(evaluationDetails.getValue()).isEqualTo(DEFAULT_VALUE);
-    assertThat(evaluationDetails.getErrorCode()).isEqualTo(ErrorCode.TARGETING_KEY_MISSING);
+    assertThat(evaluationDetails.getErrorCode()).isEqualTo(GENERAL);
     assertThat(evaluationDetails.getReason()).isEqualTo("ERROR");
     assertThat(evaluationDetails.getVariant()).isBlank();
   }
 
   @Test
   public void inconsistentTargetKey() {
```
Comment (andreas-karlsson) [inline src/test/java/com/spotify/confidence/FeatureProviderTest.java:109]: When running just this test in the editor it fails.

File: src/test/java/com/spotify/confidence/FeatureProviderTest.java (hunk 1 above)
Comment (vahidlazio) [reply src/test/java/com/spotify/confidence/FeatureProviderTest.java:99]: maybe just removing the targeting key from resolve success test and removing this one?

File: src/test/java/com/spotify/confidence/FeatureProviderTest.java (hunk 2 above)
Comment (vahidlazio) [reply src/test/java/com/spotify/confidence/FeatureProviderTest.java:109]: is it the same on master?

File: src/test/java/com/spotify/confidence/FeatureProviderTest.java [hunk 3]
```diff
@@ -271,6 +249,92 @@ public void regularResolve() {
...
+                        "prop-G",
+                        new Value(
+                            new MutableStructure(
+                                Map.of(
+                                    "prop-H", new Value(),
+                                    "prop-I", new Value())))))));
+  }
+
+  @Test
+  public void regularResolveWith2TargetingKey() {
```
Comment (nicklasl) [inline src/test/java/com/spotify/confidence/FeatureProviderTest.java:296]:
```suggestion
  public void regularResolveWith2TargetingKeyShouldPrioritiseApiOverMap() {
```

**Review by nicklasl — APPROVED** [review]
LGTM. nit suggestion about test naming.

**Review by andreas-karlsson — APPROVED** [review]

**Pull request merged**